### PR TITLE
feat: implement M4 comprehensive test coverage for Phase 3.1

### LINKED ISSUE #121: M4: Testing - Unit, integration, and E2E tests
## Parent Epic
Part of #117 (Phase 3.1 - Triage MCP Server)

## Goal
Comprehensive test coverage for all Phase 3.1 components.

---

## Tasks

| Task | Effort | Dependencies |
|------|--------|--------------|
| 4.1 Unit tests for service layer | 1.5h | M2, M3 |
| 4.2 Unit tests for MCP handlers | 1h | M2, M3 |
| 4.3 Integration tests (GitHub mock) | 1h | M3 |
| 4.4 E2E workflow test | 0.5h | M3 |

**Estimated Total:** 4 hours

---

### Unit Tests

```go
// internal/usecase/triage/service_test.go

func TestService_ListFindings(t *testing.T) {
    // Mock GitHub client returns sample comments
    // Verify filtering works correctly
    // Verify status detection from comment body
}

func TestService_GetSuggestion(t *testing.T) {
    // Mock file content
    // Verify exact code extraction
    // Verify suggestion parsing (```suggestion blocks)
    // Test edge cases: first/last lines, multi-line
}

func TestService_ParseSuggestionCode(t *testing.T) {
    // Test GitHub suggestion syntax
    // Test generic code blocks
    // Test plain text suggestions
}
```

### Integration Tests

```go
// internal/adapter/mcp/handlers/integration_test.go

func TestMCPTools_Integration(t *testing.T) {
    // Start MCP server with mocked dependencies
    // Call each tool via MCP protocol
    // Verify responses match expected format
}
```

### E2E Test

```go
// cmd/code-reviewer-mcp/e2e_test.go

func TestE2E_TriageWorkflow(t *testing.T) {
    // 1. list_findings → returns sample findings
    // 2. get_finding → returns detailed finding
    // 3. get_suggestion → returns apply-ready data
    // 4. reply_to_finding → creates comment
    // 5. mark_resolved → resolves thread
}
```

---

## Test Coverage Targets

| Package | Target Coverage |
|---------|-----------------|
| `internal/domain/` | >90% |
| `internal/usecase/triage/` | >80% |
| `internal/adapter/mcp/` | >75% |
| `internal/adapter/github/` (new code) | >75% |

---

## Deliverables

- [ ] `internal/usecase/triage/service_test.go`
- [ ] `internal/adapter/mcp/handlers/*_test.go`
- [ ] `internal/adapter/mcp/integration_test.go`
- [ ] `cmd/code-reviewer-mcp/e2e_test.go`

---

## Acceptance Criteria

- [ ] >80% code coverage on service layer
- [ ] All MCP tool handlers have unit tests
- [ ] Integration test passes with mock GitHub
- [ ] E2E workflow test demonstrates full flow
- [ ] `go test -race ./...` passes
- [ ] All tests pass: `go test ./...`
- [ ] Code formatted: `gofmt -w .`

### PR DESCRIPTION
**Pull request by bkyoung**
## Summary

- Add comprehensive unit tests for all 12 MCP handlers (0% → 81.4% coverage)
- Add integration tests verifying full triage workflow through the stack
- Add service layer tests for previously untested methods (71.7% → 85.3% coverage)

## Test plan

- [x] All unit tests pass (`go test ./...`)
- [x] Race detector passes (`go test -race ./internal/adapter/mcp/... ./internal/usecase/triage/...`)
- [x] Lint passes (`golangci-lint run`)
- [x] Build succeeds (`go build -o cr ./cmd/cr`)
- [x] Coverage targets met (MCP: 81.4% > 75%, Triage: 85.3% > 80%)

Closes #121

### CODE REVIEW

File: internal/adapter/mcp/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+			Severity:    "high",
+			Category:    "security",
+			Body:        "SQL injection vulnerability detected",
+			Author:      "github-actions[bot]",
+			IsResolved:  false,
+			ReplyCount:  0,
+		},
+	}
+
+	// Configure mock expectations
```
[the remark this answers is not in the document]
Comment (bkyoung) [reply internal/adapter/mcp/integration_test.go]: **Status:** disputed

This is intentional. The `domain.ResolveFindingID()` function strips the `CR_FP:` prefix before passing to the adapter layer (see `pr_service.go:133`). The test correctly expects the stripped fingerprint because that's what the adapter receives. The handler input uses the full ID, the service layer parses it, and the adapter sees only the raw fingerprint.

File: internal/adapter/mcp/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+func (m *MockIntegrationCommentWriter) CreateComment(ctx context.Context, owner, repo string, prNumber int, commitSHA, path string, line int, body string) (int64, error) {
+	args := m.Called(ctx, owner, repo, prNumber, commitSHA, path, line, body)
+	return args.Get(0).(int64), args.Error(1)
+}
+
+// MockIntegrationReviewManager provides mock review management operations.
+type MockIntegrationReviewManager struct {
+	mock.Mock
+}
```
Comment (bkyoung) [inline internal/adapter/mcp/integration_test.go]: **Fixed.** Added nil checks before type assertions to prevent panics when tests forget to set return values. See updated code at lines 107-110 and 115-118.

File: internal/adapter/mcp/integration_test.go [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+	mockCommentWriter := new(MockIntegrationCommentWriter)
+	mockReviewManager := new(MockIntegrationReviewManager)
+
+	// Setup test data
+	prMeta := &domain.PRMetadata{
+		Owner:   "testorg",
+		Repo:    "testrepo",
+		Number:  123,
+		HeadSHA: "abc123def",
+		BaseRef: "main",
```
[the remark this answers is not in the document]
Comment (bkyoung) [reply internal/adapter/mcp/integration_test.go]: **Status:** disputed

**Disputed** - The test correctly models the production data flow:

1. Real `PRFinding` objects have fingerprints with the `CR_FP:` prefix (e.g., `CR_FP:abc123`)
2. When `GetFinding()` is called with a full ID like `CR_FP:abc123`, the service layer calls `domain.ResolveFindingID()` which strips the prefix
3. The adapter's `GetPRCommentByFingerprint()` receives just `abc123`

Setting `findings[0].Fingerprint = "abc123"` (raw) would be **incorrect** because it wouldn't match real data. The mock expectation correctly expects the stripped value because that's what the adapter layer receives after the service processes the input.

This is the documented contract: handlers receive full IDs, service layer normalizes them, adapters work with raw values.

File: internal/adapter/mcp/integration_test.go [hunk 4]
```diff
@@ -0,0 +1,487 @@
...
+	"github.com/bkyoung/code-reviewer/internal/usecase/triage"
+)
+
+// =============================================================================
+// Integration tests: MCP Handler -> PRService -> Mocked Adapters
+// These tests verify the full stack without hitting real GitHub APIs.
+// =============================================================================
+
+// MockIntegrationAnnotationReader provides mock GitHub annotation data.
+type MockIntegrationAnnotationReader struct {
```
Comment (bkyoung) [inline internal/adapter/mcp/integration_test.go]: **Acknowledged.** Valid observation about mock reuse potential. The `MockIntegration*` types use a naming convention to distinguish integration test mocks from unit test mocks, even though they're functionally similar. This is a style choice for test organization clarity. Consolidating would reduce lines but might blur the distinction between unit and integration test fixtures. Will consider refactoring if the test suite grows significantly.
